gadgets/traceloop/go/program.go

```diff
@@ -0,0 +1,824 @@
+	"mount":       {useNullByteLength, useNullByteLength, useNullByteLength, 0, 0, 0},
+	"umount2":     {useNullByteLength, 0, 0, 0, 0, 0},
+	"sethostname": {useNullByteLength, 0, 0, 0, 0, 0},
+	"statfs":      {useNullByteLength, 0, 0, 0, 0, 0},
+	"stat":        {useNullByteLength, 0, 0, 0, 0, 0},
+	"statx":       {0, useNullByteLength, 0, 0, 0, 0},
+	"lstat":       {useNullByteLength, 0, 0, 0, 0, 0},
+	"fgetxattr":   {0, useNullByteLength, 0, 0, 0, 0},
+	"lgetxattr":   {useNullByteLength, useNullByteLength, 0, 0, 0, 0},
+	"getxattr":    {useNullByteLength, useNullByteLength, 0, 0, 0, 0},
+	"newfstatat":  {0, useNullByteLength, 0, 0, 0, 0},
+	"read":        {0, useRetAsParamLength | paramProbeAtExitMask, 0, 0, 0, 0},
+	"write":       {0, useArgIndexAsParamLength + 2, 0, 0, 0, 0},
+	"getcwd":      {useNullByteLength | paramProbeAtExitMask, 0, 0, 0, 0, 0},
+	"pread64":     {0, useRetAsParamLength | paramProbeAtExitMask, 0, 0, 0, 0},
+}
+
+var (
+	dsOutput api.DataSource
+	fields   map[string]api.Field
+)
+
+type containerRingReader struct {
+	innerBuffer api.Map
+	perfReader  api.PerfReader
+	mntnsID     uint64
+}
+
+type tracelooper struct {
+	mapOfPerfBuffers api.Map
+
+	// key:   cgroupID
+	// value: *containerRingReader
+	readers sync.Map
+}
+
+var t tracelooper
+
+// Keep in sync with type in program.bpf.c.
+type traceloopSyscallEventContT struct {
+	EventType          uint8
+	Param              [128]uint8
+	MonotonicTimestamp uint64
+	Length             uint64
+	Index              uint8
+	Failed             uint8
+	_                  [5]byte
+}
+
+// Keep in sync with type in program.bpf.c.
+type traceloopSyscallEventT struct {
+	EventType          uint8
+	Args               [6]uint64
+	MonotonicTimestamp uint64
+	BootTimestamp      uint64
+	Pid                uint32
+	Cpu                uint16
+	Id                 uint16
+	Comm               [16]uint8
+	ContNr             uint8
+	Typ                uint8
+	_                  [61]byte
+}
+
+type syscallEvent struct {
+	bootTimestamp      uint64
+	monotonicTimestamp uint64
+	typ                uint8
+	contNr             uint8
+	cpu                uint16
+	id                 uint16
+	pid                uint32
+	comm               string
+	args               []uint64
+	mountNsID          uint64
+	retval             uint64
+}
+
+type syscallEventContinued struct {
+	monotonicTimestamp uint64
+	index              uint8
+	param              string
+}
+
+type syscallParam struct {
+	name    string
+	value   string
+	content *string
+}
+
+type event struct {
+	timestamp  int64
+	mountNsID  uint64
+	cpu        uint16
+	pid        uint32
+	comm       string
+	syscall    string
+	parameters []syscallParam
+	retval     string
+}
+
+func toString(parameters []syscallParam) string {
+	var sb strings.Builder
+
+	for idx, p := range parameters {
+		value := p.value
+		if p.content != nil {
+			value = *p.content
+		}
+		sb.WriteString(fmt.Sprintf("%s=%s", p.name, value))
```

Comment:
perhaps this is a bit more efficient:

```suggestion
	        sb.WriteString(p.name)
                sb.WriteRune('=')
                sb.WriteString(p.value)
```